src/components/filter-fields/DatePicker2.jsx

```diff
@@ -0,0 +1,278 @@
+                  <ChevronDown className="w-4 h-6" />
+                </button>
+                <div className="dropdown-content">
+                  <table>
+                    <tbody>
+                      {createMonths().map((row) => (
+                        <tr key={row[0]}>
+                          {row.map((monthName) => (
+                            <td key={monthName} onClick={() => formik.setFieldValue('month', monthName)}>
+                              {monthName}
+                            </td>
+                          ))}
+                        </tr>
+                      ))}
+                    </tbody>
+                  </table>
+                </div>
+              </div>
+              <div className="input-year">
+                <input
+                  className="inputYear"
+                  required
+                  type="text"
+                  size="4"
+                  placeholder="XXXX"
+                  id="year"
+                  name="year"
+                  onChange={formik.handleChange}
+                  value={formik.values.year}
+                />
+                {formik.errors.year && formik.touched.year && (
+                  <p className="errorMessage">{formik.errors.year}</p>
+                )}
+              </div>
+              <div className="dropdown-date">
+                <button type="button" className="dropbtn">
+                  <ChevronDown className="w-4 h-6" />
+                </button>
+                <div className="dropdown-content" style={{ alignItems: 'center' }}>
+                  <div className="chevron"><ChevronLeft onClick={prevPage} className="w-6 h-6 mt+5 ml-2" /></div>
+                  <div className="yearsTable">
+                    <table>
+                      <tbody>
+                        {createYears().map((row) => (
+                          <tr key={row[0]}>
+                            {row.map((yearNumber) => (
+                              <td key={yearNumber} onClick={() => formik.setFieldValue('year', yearNumber)}>
+                                {yearNumber}
+                              </td>
+                            ))}
+                          </tr>
+                        ))}
+                      </tbody>
+                    </table>
+                  </div>
+                  <div className="chevron"><ChevronRight onClick={nextPage} className="w-6 h-6 mt+5 ml-2" /></div>
+                </div>
+              </div>
+            </div>
+            <div className="button-container">
+              <button type="button" onClick={onClear} className="cancelButton">{t('cancel')}</button>
+              <button type="submit" className="okButton">OK</button>
+            </div>
+          </div>
+        </Form>
+      </div>
+    </>
+  );
+};
+
+DatePicker2.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
```

Comment:
Your components not react for `value` prop changed.
PropTypes.string, PropTypes.array??